packages/malloy/src/lang/test/parse.spec.ts

```diff
@@ -539,20 +1022,35 @@ describe('qops', () => {
   });
   test('index sample-rows', modelOK('query:a->{index: *; sample: 100000}'));
   test('top N', modelOK('query: a->{ top: 5; group_by: astr }'));
-  test('top N by field', modelOK('query: a->{top: 5 by af; group_by: astr}'));
+  test('top N by field', modelOK('query: a->{top: 5 by astr; group_by: astr}'));
   test(
     'top N by expression',
-    modelOK('query: ab->{top: 5 by acount; group_by: astr}')
+    modelOK('query: ab->{top: 5 by ai + 1; group_by: ai}')
   );
+  test('top N by field must be in the output space', () =>
+    expect('query: a->{top: 5 by af; group_by: astr}').compileToFailWith(
+      'Unknown field af in output space'
+    ));
   test('limit N', modelOK('query: a->{ limit: 5; group_by: astr }'));
-  test('order by', modelOK('query: a->{ order_by: af; group_by: astr }'));
+  test('order by', modelOK('query: a->{ order_by: astr; group_by: astr }'));
+  test(
+    'order by preserved over refinement',
+    modelOK(`
+      query: a1 is a -> { group_by: astr }
+      query: -> a1 { order_by: astr }
+    `)
+  );
+  test('order by must be in the output space', () =>
+    expect('query: a -> { order_by: af; group_by: astr }').compileToFailWith(
+      'Unknown field af in output space'
+    ));
   test(
     'order by asc',
-    modelOK('query: a->{ order_by: af asc; group_by: astr }')
+    modelOK('query: a->{ order_by: astr asc; group_by: astr }')
   );
   test(
     'order by desc',
-    modelOK('query: a->{ order_by: af desc; group_by: astr }')
+    modelOK('query: a->{ order_by: astr desc; group_by: astr }')
```

Comment:
We didn't require in the translator that order by fields be in the output space, though if they weren't we generated probably incorrect SQL. Now we do, so I updated these tests.